orga/app/components/auth/login-or-register.js

```diff
@@ -1,12 +1,28 @@
 import { action } from '@ember/object';
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
 
 export default class LoginOrRegister extends Component {
+  @service currentDomain;
+  @service locale;
+  @service intl;
+  @service router;
   @tracked displayRegisterForm = true;
+  @tracked selectedLanguage = this.intl.primaryLocale;
 
   @action
   toggleFormsVisibility() {
     this.displayRegisterForm = !this.displayRegisterForm;
   }
+
+  @action
+  onLanguageChange(value) {
+    this.selectedLanguage = value;
+    this.locale.setLocale(this.selectedLanguage);
```

Comment:
**Suggestion** : Dans le cas où l'utilisateur arrive sur la page avec le paramètre ?lang=en, il manque le code pour enlever ce paramètre de l'url lorsqu'on change la langue avec le language switcher en français. 
Je l'ai fait de cette façon dans le ticket côté certif : 
```suggestion
    this.locale.setLocale(this.selectedLanguage);
    this.router.replaceWith('join', {
      queryParams: {
        code: this.args.certificationCenterInvitationCode,
        invitationId: this.args.certificationCenterInvitationId,
        lang: null,
      },
```

Reply:
Je modifie !

Reply:
Sur orga, c'est pas possible de taper "http://localhost:4201/rejoindre?invitationId=100057&code=INVIABC123?lang=en" dans l'url, on est redirigé vers connexion. J'ai mis un test d'acceptance là-dessus.

Du coup, je sais pas trop quoi faire. 
Faudrait que je change le code pour permettre "http://localhost:4201/rejoindre?invitationId=100057&code=INVIABC123?lang=en"  et d'ajouter le code qui transforme ça en "http://localhost:4201/rejoindre?invitationId=100057&code=INVIABC123"  ?
 Any idea ? On voit avec Emmy ?

Reply:
Done

Reply:
En effet ! J'ai regardé de mon côté pour trouver une solution là-dessus et ce bout de code devrait fonctionner : 
```suggestion
    this.locale.setLocale(this.selectedLanguage);
    this.router.replaceWith('join', {
      queryParams: {
        lang: null,
      },
    });
```
Le seul "problème" est que lorsque t'arrives sur la page avec une url [http://localhost:4201/rejoindre?invitationId=100057&code=INVIABC123&lang=en](http://localhost:4201/rejoindre?invitationId=100057&code=INVIABC123&lang=en), l'ordre des queryParams est modifiée pour avoir un ordre alphabétique. Cela donnera `.../rejoindre?code=INVIABC123&invitationId=100057`.

A voir si s'il y a une meilleure solution 😄